IOMC/ParticleGuns/src/CloseByParticleGunProducer.cc

```diff
@@ -0,0 +1,125 @@
+#include <ostream>
+
+#include "IOMC/ParticleGuns/interface/CloseByParticleGunProducer.h"
+
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+#include "SimDataFormats/GeneratorProducts/interface/GenEventInfoProduct.h"
+
+#include "DataFormats/Math/interface/Vector3D.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "CLHEP/Random/RandFlat.h"
+#include "CLHEP/Units/GlobalSystemOfUnits.h"
+#include "CLHEP/Units/GlobalPhysicalConstants.h"
+#include "CLHEP/Random/RandFlat.h"
+
+using namespace edm;
+using namespace std;
+
+CloseByParticleGunProducer::CloseByParticleGunProducer(const ParameterSet& pset) :
+   BaseFlatGunProducer(pset)
+{
+
+  ParameterSet defpset ;
+  ParameterSet pgun_params =
+    pset.getParameter<ParameterSet>("PGunParameters") ;
+
+  fEn = pgun_params.getParameter<double>("En");
+  fR = pgun_params.getParameter<double>("R");
+  fZ = pgun_params.getParameter<double>("Z");
+  fDelta = pgun_params.getParameter<double>("Delta");
+  fPartIDs = pgun_params.getParameter< vector<int> >("PartID");
+  fPointing = pgun_params.getParameter<bool>("Pointing");
+
```

Comment:
do you have an example configuration fragment usable for tests?

Reply:
I have, since I already privately used it extensively. Would like me to put the snippet somewhere in this package? If so, where exactly?

This is an example:

```python
process.generator = cms.EDProducer("CloseByParticleGunProducer",
                                   PGunParameters = cms.PSet(PartID = cms.vint32(22, 22),
                                                             En = cms.double(50.),
                                                             R = cms.double(60),
                                                             Z = cms.double(320),
                                                             Delta = cms.double(2.5),
                                                             Pointing = cms.bool(True),
                                                             MaxEta = cms.double(2.9),
                                                             MaxPhi = cms.double(3.14159265359/6.),
                                                             MinEta = cms.double(1.5),
                                                             MinPhi = cms.double(-3.14159265359/6.),
                                                             ),
                                   Verbosity = cms.untracked.int32(10),
                                   psethack = cms.string('two close by particles'),
                                   AddAntiParticle = cms.bool(False),
                                   firstRun = cms.untracked.uint32(1)
                                   )

```

Reply:
We normally store these default fragments in Configuration/Generator/python , where they may be used for instance by RelVals .